nodejs-polars/src/conversion/from.rs

```diff
@@ -116,6 +118,62 @@ impl FromJsUnknown for AnyValue<'_> {
     }
 }
 
+impl FromJsUnknown for DataType {
+    fn from_js(val: JsUnknown) -> Result<Self> {
+        match val.get_type()? {
+            ValueType::Undefined | ValueType::Null => Ok(DataType::Null),
+            ValueType::Boolean => Ok(DataType::Boolean),
+            ValueType::Number => Ok(DataType::Float64),
+            ValueType::String => Ok(DataType::Utf8),
+            ValueType::Bigint => Ok(DataType::UInt64),
+            ValueType::Object => {
+                if val.is_date()? {
+                    Ok(DataType::Datetime(TimeUnit::Milliseconds, None))
+                } else if val.is_array()? {
+                    Vec::<DataType>::from_js(val)
+                        .map(|list| DataType::List(Box::new(coerce_data_type(&list))))
+                } else {
+                    Ok(DataType::Utf8)
+                }
+            }
+            _ => panic!("not supported"),
+        }
+    }
+}
+
+fn coerce_data_type<A: Borrow<DataType>>(datatypes: &[A]) -> DataType {
```

Comment:
Maybe use the [https://github.com/pola-rs/polars/blob/9c82106cba1dd9f31e4e7893020dfc388742fd6b/polars/polars-core/src/utils/mod.rs#L374] for DRYness?

Reply:
Oh nice, I didn't realize that was there.